addons/web/static/src/xml/web_calendar.xml

```diff
@@ -12,7 +12,7 @@
     </div>
 
     <t t-name="calendar-box">
-        <div t-att-style="typeof color === 'string' ? ('background-color:' + color) + ';' : ''" t-attf-class="#{record.is_highlighted ? 'o_event_hightlight' : ''} #{typeof color === 'number' ? _.str.sprintf('o_calendar_color_%s', color) : 'o_calendar_color_1'}">
+        <div t-att-style="typeof color === 'string' ? ('background-color:' + color) + ';' : ''" t-attf-class="#{record.is_highlighted ? 'o_event_hightlight' : ''} #{typeof color === 'number' ? _.str.sprintf('o_calendar_color_%s', color) : 'o_calendar_color_1'} #{record.is_hatched ? 'o_calendar_hatched o_calendar_hatched_' : ''}#{typeof color === 'number' ? color : 1}">
```

Comment:
You need to include the `typeof(color) ...` in the is_hatched condition, otherwise you might get a single number as class which shouldn't be there.

Reply:
Done